test/031_delegable.ts

```diff
@@ -0,0 +1,77 @@
+import DelegableArtifact from '../artifacts/contracts/test/DelegableMock.sol/DelegableMock.json'
+import { DelegableMock as Delegable } from '../typechain/DelegableMock'
+
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address'
+// @ts-ignore
+import { ethers, waffle } from 'hardhat'
+const { deployContract } = waffle
+import { expect } from 'chai'
+
+
+describe('Delegable', () => {
+  let delegable: Delegable
+  let delegableFromOther: Delegable
+  let ownerAcc: SignerWithAddress
+  let owner: string
+  let otherAcc: SignerWithAddress
+  let other: string
+
+  before(async () => {
+    const signers = await ethers.getSigners()
+    ownerAcc = signers[0]
+    owner = await ownerAcc.getAddress()
+    otherAcc = signers[1]
+    other = await otherAcc.getAddress()
+
+    delegable = (await deployContract(signers[0], DelegableArtifact, [])) as Delegable
+    delegableFromOther = delegable.connect(otherAcc)
+  })
+
+  it('allows calling restricted functions to account owner', async () => {
+    await delegable.restricted(owner)
+  })
+
+  it('does not allow calling restricted functions to others', async () => {
+    await expect(
+      delegableFromOther.restricted(owner)
+    ).to.be.revertedWith(
+        'DelegableMock: Forbidden'
+    )
+  })
+
+  it('does not allow revoking delegations that do not exist', async () => {
+    await expect(
+      delegable.revokeDelegate(other)
+    ).to.be.revertedWith(
+        'Delegable: Already undelegated'
+    )
+  })
+
+  it('adds delegates', async () => {
+    await delegable.addDelegate(other)
+    expect(await delegable.delegated(owner, other)).to.be.true
+  })
+
+  describe('with delegates', async () => {
+    beforeEach(async () => {
+      // await delegable.addDelegate(other) <-- Keeps state from 'adds delegates'
```

Comment:
Why does this not revert the state between tests?

Reply:
You need to add a `beforeEach` on L16 instead of `before`